pkg/controller/sync/controller.go

```diff
@@ -957,7 +957,7 @@ func (s *FederationSyncController) objectForCluster(template, override *unstruct
 	overrideField := overridePath[len(overridePath)-1]
 	for _, overrideInterface := range overrides {
 		clusterOverride := overrideInterface.(map[string]interface{})
-		if clusterOverride["clustername"] != clusterName {
+		if clusterOverride["clusterName"] != clusterName {
```

Comment:
If you update here, the `federated<Type>-override.yaml` at https://github.com/kubernetes-sigs/federation-v2/tree/master/example/sample1 should also be updated.

```
  spec:
    overrides:
    - clustername: cluster2 <<<<
      data:
        foo: bar
```

The commit message is also not accurate, you are not only changing `clusternames` but also `clustername`.

Another point is it is terrible that the test `**override.yaml` are using `clusterName`, but the template in example are using `clustername`, why not keep the files in `sample1` and `fixtures` keep synced and use same file?

Reply:
Good catch on both clustername and clusternames needing to be fixed.  I disagree that the tests need to match the samples, they serve different use case.  The goal of ensuring consistent field name usage can be accomplished in code as per the new commit in this PR.